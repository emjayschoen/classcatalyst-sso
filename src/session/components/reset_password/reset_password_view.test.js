import React from 'react';
import { shallow } from 'enzyme/build';
import ResetPasswordView from './reset_password_view';

describe('<ResetPasswordView />', () => {
  let subject;

  describe('not logged in', () => {
    beforeEach(() => {
      subject = shallow(<ResetPasswordView signedIn={false} handleSubmit={jest.fn()} />);
    });

    it('renders correctly', () => {
      expect(subject).toMatchSnapshot();
    });
  });

  describe('when already logged in', () => {
    beforeEach(() => {
      subject = shallow(<ResetPasswordView signedIn handleSubmit={jest.fn()} />);
    });

    it('redirects to /', () => {
      expect(subject).toMatchSnapshot();
    });
  });
});
