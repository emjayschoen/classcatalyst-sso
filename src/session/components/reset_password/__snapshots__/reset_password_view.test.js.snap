// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPasswordView /> not logged in renders correctly 1`] = `
<styled.div>
  <h3>
    Reset your password
  </h3>
  <form
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      fullWidth={true}
      label="New Password"
      name="password"
      required={true}
      type="password"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Confirm New Password"
      name="password_confirmation"
      required={true}
      type="password"
    />
    <input
      className="btn"
      disabled={false}
      type="submit"
      value="Reset Password"
    />
    <span
      className="status"
    />
  </form>
</styled.div>
`;

exports[`<ResetPasswordView /> when already logged in redirects to / 1`] = `
<Redirect
  to="/"
/>
`;
